src/loans/solend.py

```diff
@@ -45,39 +50,379 @@ def get_events(start_block_number: int = 0) -> pandas.DataFrame:
+                self.loan_entities[obligation] = SolendLoanEntity(obligation, last_update)
+            # Process collateral positions
+            collateral_data = row['collateral']
+            for mint, collateral_info in collateral_data.items():
+                if collateral_info:  # Check if there is data present for the collateral
+                    new_collateral_position = SolendCollateralPosition(
+                        reserve=collateral_info['reserve'],
+                        mint=mint,
+                        amount=collateral_info['amount'],
+                    )
+                    self.loan_entities[obligation].collateral.append(new_collateral_position)
+
+            # Process debt positions
+            debt_data = row['debt']
+            for mint, debt_info in debt_data.items():
+                if debt_info:  # Check if there is data present for the debt
+                    new_debt_position = SolendDebtPosition(
+                        reserve=debt_info['reserve'],
+                        mint=mint,
+                        raw_amount=debt_info['rawAmount'],
+                    )
+                    self.loan_entities[obligation].debt.append(new_debt_position)
+
+    def _get_reserve_configs(self):
+        reserve_addresses = {
+            reserve
+            for loan_entity in self.loan_entities.values()
+            for reserve in loan_entity.get_unique_reserves()
+        }
+        ids = ",".join(reserve_addresses)
+        url = f'https://api.solend.fi/v1/reserves/?ids={ids}'
+
+        response = requests.get(url, timeout=15)
+        self.reserve_configs = {
+            reserve['reserve']['address']: reserve
+            for reserve in response.json()['results']
+        }
+
+    def save_health_ratios(self) -> None:
+        """
+        Compute and save health ratios to the database.
+        """
+        if not self.reserve_configs:
+            self._get_reserve_configs()
+        with db.get_db_session() as session:
+            for loan_entity in self.loan_entities.values():
+                loan_entity.update_positions_from_reserve_config(self.reserve_configs)
+                new_health_ratio = db.SolendHealthRatio(
+                    slot=int(self.last_slot),
+                    user=loan_entity.obligation,
+                    health_factor=loan_entity.health_ratio(),
+                    std_health_factor=loan_entity.std_health_ratio(),
+                    collateral=str(round(loan_entity.collateral_market_value(), 6)),
+                    risk_adjusted_collateral=str(round(loan_entity.collateral_risk_adjusted_market_value(), 6)),
+                    debt=str(round(loan_entity.debt_market_value(), 6)),
+                    risk_adjusted_debt=str(round(loan_entity.debt_risk_adjusted_market_value(), 6))
+                )
+                session.add(new_health_ratio)
+            session.commit()
+
+    def find_relevant_mint_collateral_pairs(self):
```

Comment:
What do you mean by mint collateral pairs? Shouldn't it be debt-collateral pairs?

Reply:
exactly. fixed